src/Microsoft.Health.Fhir.CosmosDb/Features/Search/CosmosDbSortingValidator.cs

```diff
@@ -26,7 +36,16 @@ public bool ValidateSorting(IReadOnlyList<(SearchParameterInfo searchParameter,
                     errorMessages = Array.Empty<string>();
                     return true;
                 case { Count: 1 }:
-                    errorMessages = new[] { string.Format(CultureInfo.InvariantCulture, Core.Resources.SearchSortParameterNotSupported, sorting[0].searchParameter.Code) };
+                    (SearchParameterInfo searchParameter, SortOrder sortOrder) parameter = sorting[0];
+
+                    if (parameter.searchParameter.SortStatus == SortParameterStatus.Enabled ||
+                        (parameter.searchParameter.SortStatus == SortParameterStatus.Supported && _contextAccessor.FhirRequestContext?.RequestHeaders.TryGetValue(KnownHeaders.PartiallyIndexedParamsHeaderName, out StringValues _) == true))
```

Comment:
ContainsKey()? Or do we want to check for a specific value?

Reply:
Assuming `_contextAccessor.FhirRequestContext?.RequestHeaders.TryGetValue(KnownHeaders.PartiallyIndexedParamsHeaderName, out StringValues _)`? This is the same check used elsewhere for the search parameters, I'll group this code into an extension so it remains consistent